js-exercises/update-inventory/updateInventory.js

```diff
@@ -0,0 +1,39 @@
+function updateInventory(...args) {
+  const originalArr = args[0].map(element => element);
+  const newArr = args[1];
+  const result = [];
+  const elementIndexMap = new Map();
+  originalArr.forEach((element, index) => {
+    const itemName = element[1];
+    elementIndexMap.set(itemName, index);
+  });
+  newArr.forEach((element) => {
+    const quantity = element[0];
+    const itemName = element[1];
+    const indexInOriginalArr = elementIndexMap.get(itemName);
+    if (indexInOriginalArr !== undefined) {
+      const oldElement = originalArr[indexInOriginalArr];
+      const oldQuantity = oldElement[0];
+      const newQuantity = oldQuantity + quantity;
+      oldElement[0] = newQuantity;
```

Comment:
1. Its a very very minor one. Since you are mutating a variable I feel its better to use let instead of const even though you can mutate an array since its not a primitive type.